esx_service/vmdk_ops.py

```diff
@@ -837,12 +837,32 @@ def parse_vol_name(full_vol_name):
             raise ValidationError("Volume name contains illegal characters: {0}".format(c))
 
     # Check if the datastore name is too long
-    if ds_name and len(ds_name) > MAX_DS_NAME_LEN:
-        raise ValidationError("Datastore name is too long (max len is {0})".format(MAX_DS_NAME_LEN))
+    if ds_name:
+        if len(ds_name) > MAX_DS_NAME_LEN:
+            raise ValidationError("Datastore name is too long (max len is {0})".format(MAX_DS_NAME_LEN))
+
+        matching_datastores = find_matching_datastores(ds_name)
```

Comment:
nit (feel free to ignore): it's probably easier to say 
```python
    matching_datastores = [d for d in get_datastore_names_list() if d.lower() == ds_name.lower()]
``` 
and drop find_matching_datastores(ds_name) function all together.